src/node/interfaces.cpp

```diff
@@ -409,6 +410,7 @@ class NodeImpl : public Node
     std::unique_ptr<Handler> handleNotifyBlockTip(NotifyBlockTipFn fn) override
     {
         return MakeSignalHandler(::uiInterface.NotifyBlockTip_connect([fn, this](SynchronizationState sync_state, const CBlockIndex* block) {
+            LOCK(chainman().GetMutex());
```

Comment:
Am I correct in understanding that `handleNotifyBlockTip` takes a lock now where it didn't before?